aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,623 @@
+///     two types of proposals can be created:
+///     3.a: no-op, no automatic execution happens on chain. Only the proposal and its results are recorded on-chain for DAO admin to take actions
+///     3.b: transfer fund, the fund will be automatically transferred from DAO resource account to the destination account.
+/// 4. A voter can vote for a proposal of a DAO through `vote`
+/// 5. Anyone can call the `resolve` to resolve a proposal. A proposal voting duration has to expire and the proposal should have more votes than the minimal required threshold.
+///
+/// An example of DAO delegation flow
+///  1. the DAO creator can delegate her DAO through `delegate_dao` to another admin address
+///
+/// An example of DAO removal.
+/// 1. The DAO creator can call `reclaim_signer_capability` to remove her DAO from the platform and get back her resource account's signercapability
+module dao_platform::nft_dao {
+    use aptos_framework::account::{SignerCapability, create_signer_with_capability, get_signer_capability_address};
+    use aptos_framework::account;
+    use aptos_framework::aptos_coin::AptosCoin;
+    use aptos_framework::coin;
+    use aptos_framework::timestamp;
+    use aptos_std::bucket_table::BucketTable;
+    use aptos_std::bucket_table;
+    use aptos_std::table::Table;
+    use aptos_std::table;
+    use aptos_token::property_map::PropertyMap;
+    use aptos_token::property_map;
+    use aptos_token::token::{Self, TokenId};
+    use dao_platform::nft_dao_events::emit_create_dao_event;
+    use dao_platform::token_type::get_token_type;
+    use std::bcs;
+    use std::error;
+    use std::signer;
+    use std::string::String;
+    use std::string;
+    use std::vector;
+
+    /// The new address to be delegated to already exists in the platform
+    const EDELEGATED_ADDRESS_ALREADY_EXISTS: u64 = 1;
+
+    /// This account doesn't have enough voting weights
+    const EVOTING_WEIGHTS_NOT_ENOUGH: u64 = 2;
+
+    /// This account doesn't own this DAO's voting token
+    const ENOT_OWN_THE_VOTING_DAO_TOKEN: u64 = 3;
+
+    /// This function is supported in proposal
+    const ENOT_SUPPROTED_FUNCTION: u64 = 4;
+
+    /// Can only propose a start time in future
+    const EPROPOSAL_START_TIME_SHOULD_IN_FUTURE: u64 = 5;
+
+    /// Cannot only use membership token of the DAO
+    const ENOT_FOUND_ADMIN_ACCOUNT: u64 = 6;
+
+    /// String length exceeds limits
+    const ESTRING_TOO_LONG: u64 = 7;
+
+    /// Proposal ended
+    const EPROPOSAL_ENDED: u64 = 8;
+
+    /// Propsoal already resolved
+    const EPROPOSAL_RESOLVED: u64 = 9;
+
+
+    /// TOKEN VOTED FOR PROPOSAL
+    const ETOKEN_ALREADY_VOTED: u64 = 10;
+
+    /// Resource Account Doesn't exist
+    const ERESOURCE_ACCT_NOT_EXIST: u64 = 11;
+
+    /// Proposal resource not created
+    const EPRPOSALS_NOT_EXIST_AT_ADDRESS: u64 = 12;
+
+    /// Proposal id doesn't exist
+    const EPRPOSAL_ID_NOT_EXIST: u64 = 13;
+
+
+    /// Constants
+    const PROPOSAL_UNRESOLVED: u8 = 0;
```

Comment:
nit: Name this PROPOSAL_PENDING instead as PROPOSAL_UNRESOLVED can mean the proposal's voting has ended but it has not been resolved yet.